File: LeetCode/1588.cpp
//Link: https://leetcode.com/problems/sum-of-all-odd-length-subarrays/

class Solution {
public:
    int sumOddLengthSubarrays(vector<int>& arr) {
        int e, n = arr.size(), sum = 0;
        for(int s = 0 ; s < n ; s++) {
            for(int k = 1 ; s+k <= n ; k += 2) {
                e = s+k;
                for(int i = s ; i < e ; i++)
                    sum += arr[i];
            }
        }
        return sum;
    }
};

File: LeetCode/py/1588.py
#Link: https://leetcode.com/problems/sum-of-all-odd-length-subarrays/

class Solution:
    def sumOddLengthSubarrays(self, arr: List[int]) -> int:
        total = 0
        for s in range(len(arr)):
            k = 1
            while s+k <= len(arr):
                total += sum(arr[s:s+k])
                k += 2
        return total
        